### ISSUE TITLE
Model conversion fails with cryptic error when feeding sample during post-training quantization

### ISSUE THREAD

**Issue by GPhilo** [tensorflow/tensorflow#43239]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (or github SHA if from source): Nightly 2.4.0-dev20200914

**Command used to run the converter or code if you’re using the Python API**

```python
import numpy as np
import tensorflow as tf
import cv2

def representative_dataset_gen(input_image_shape, num_samples_to_generate=100): # dummy generator
  h, w, *_ = input_image_shape
  for _ in range(num_samples_to_generate):
    im = np.random.random([w,h,3])
    yield [im.astype(np.float32)]


# Convert the model
converter = tf.lite.TFLiteConverter.from_saved_model('saved_model') # from the attached zip file
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8,
                                       tf.lite.OpsSet.SELECT_TF_OPS]
converter.inference_input_type = tf.int8
#converter.inference_output_type = tf.int8
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.representative_dataset = lambda : representative_dataset_gen([256, 1024, 3], 5)
tflite_model = converter.convert()
```

**The output from the converter invocation**

```
2020-09-15 15:51:43.315246: W tensorflow/compiler/mlir/lite/python/tf_tfl_flatbuffer_helpers.cc:315] Ignored output_format.
2020-09-15 15:51:43.315283: W tensorflow/compiler/mlir/lite/python/tf_tfl_flatbuffer_helpers.cc:318] Ignored drop_control_dependency.
2020-09-15 15:51:43.315515: I tensorflow/cc/saved_model/reader.cc:32] Reading SavedModel from: models/runs/mobilenet_v2_kitti_1024x256/tflite/saved_model
2020-09-15 15:51:43.373628: I tensorflow/cc/saved_model/reader.cc:55] Reading meta graph with tags { serve }
2020-09-15 15:51:43.373665: I tensorflow/cc/saved_model/reader.cc:93] Reading SavedModel debug info (if present) from: models/runs/mobilenet_v2_kitti_1024x256/tflite/saved_model
2020-09-15 15:51:43.373707: I tensorflow/compiler/jit/xla_gpu_device.cc:99] Not creating XLA devices, tf_xla_enable_xla_devices not set
2020-09-15 15:51:43.560665: I tensorflow/cc/saved_model/loader.cc:190] Restoring SavedModel bundle.
2020-09-15 15:51:44.001062: I tensorflow/cc/saved_model/loader.cc:174] Running initialization op on SavedModel bundle at path: models/runs/mobilenet_v2_kitti_1024x256/tflite/saved_model
2020-09-15 15:51:44.187305: I tensorflow/cc/saved_model/loader.cc:261] SavedModel load for tags { serve }; Status: success: OK. Took 871790 microseconds.
see current operation: %126 = "tfl.concatenation"(%125#0, %125#1, %125#2, %125#3) {axis = 1 : i32, fused_activation_function = "NONE"} : (tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>) -> tensor<100x4xf32>
2020-09-15 15:51:46.541754: W tensorflow/compiler/mlir/lite/tf_to_tfl_flatbuffer.cc:138] see current operation: %126 = "tfl.concatenation"(%125#0, %125#1, %125#2, %125#3) {axis = 1 : i32, fused_activation_function = "NONE"} : (tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>) -> tensor<100x4xf32>

see current operation: %129 = "tfl.concatenation"(%128#0, %128#1, %128#2, %128#3) {axis = 1 : i32, fused_activation_function = "NONE"} : (tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>) -> tensor<100x4xf32>
2020-09-15 15:51:46.541786: W tensorflow/compiler/mlir/lite/tf_to_tfl_flatbuffer.cc:138] see current operation: %129 = "tfl.concatenation"(%128#0, %128#1, %128#2, %128#3) {axis = 1 : i32, fused_activation_function = "NONE"} : (tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>) -> tensor<100x4xf32>

see current operation: %140 = "tfl.concatenation"(%133, %137, %135, %139) {axis = 1 : i32, fused_activation_function = "NONE"} : (tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>) -> tensor<100x4xf32>
2020-09-15 15:51:46.541823: W tensorflow/compiler/mlir/lite/tf_to_tfl_flatbuffer.cc:138] see current operation: %140 = "tfl.concatenation"(%133, %137, %135, %139) {axis = 1 : i32, fused_activation_function = "NONE"} : (tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>, tensor<100x1xf32>) -> tensor<100x4xf32>

2020-09-15 15:51:46.618251: I tensorflow/compiler/jit/xla_gpu_device.cc:99] Not creating XLA devices, tf_xla_enable_xla_devices not set
INFO: TfLiteFlexDelegate delegate: 3 nodes delegated out of 143 nodes with 1 partitions.

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/slr/anaconda3/envs/tf_nightly/lib/python3.8/site-packages/tensorflow/lite/python/lite.py", line 724, in convert
    return super(TFLiteSavedModelConverterV2,
  File "/home/slr/anaconda3/envs/tf_nightly/lib/python3.8/site-packages/tensorflow/lite/python/lite.py", line 648, in convert
    result = self._calibrate_quantize_model(result, **flags)
  File "/home/slr/anaconda3/envs/tf_nightly/lib/python3.8/site-packages/tensorflow/lite/python/lite.py", line 474, in _calibrate_quantize_model
    return calibrate_quantize.calibrate_and_quantize(
  File "/home/slr/anaconda3/envs/tf_nightly/lib/python3.8/site-packages/tensorflow/lite/python/optimize/calibrator.py", line 94, in calibrate_and_quantize
    self._calibrator.FeedTensor(sample)
IndexError: _Map_base::at
```

**Also, please include a link to the saved model or GraphDef**

[saved_model.zip](https://github.com/tensorflow/tensorflow/files/5225443/saved_model.zip)

**Failure details**
When converting the model in FP32 mode, the model converts and works correctly.
When attempting post-training-quantization, a cryptic error message is raised with no information regarding what exactly went wrong and making it impossible to debug any further.

**Comment by amahendrakar** [comment]
@GPhilo,
The `saved_model.zip` file you have provided seems to be broken. I have tried extracting the file in multiple machines and it doesn't work. 

Could you please create another zip archive or try uploading it as a tar file? Thanks!

**Comment by GPhilo** [comment]
Of course! Here's a tar.xz file. GitHub "doesn't support" the file type, so I appended a ".zip" extension. **Please remove the extra .zip extension after downloading.**
[saved_model.tar.xz.zip](https://github.com/tensorflow/tensorflow/files/5231469/saved_model.tar.xz.zip)

**Comment by amahendrakar** [comment]
Was able to reproduce the issue.

On running the code with [TF v2.3](https://colab.research.google.com/gist/amahendrakar/07b7a59cfa02165a94fba2bda826ebcd/43239.ipynb), Colab notebook crashes. Whereas, running the code with [TF-nightly](https://colab.research.google.com/gist/amahendrakar/c913169ac053463ea790317940d24f70/43239-tf-nightly.ipynb#scrollTo=5Dj-ZTZDmzPr) throws an error stating `IndexError: _Map_base::at`. Please find the attached gist. Thanks!

**Comment by jvishnuvardhan** [comment]
@GPhilo Looks like there may be some issue with `representative_dataset_gen` in the code.  When I tried to convert to simple tflite float model, it is converting successfully. Only when `representative_dataset_gen` enables, it is throwing an error. 

Can you please share model building code. If proprietary, then create a simple model with public data to demonstrate the issue. Thanks!

**Comment by GPhilo** [comment]
@jvishnuvardhan Indeed, that was also the behaviour I reported in the issue description above.
I prepared a [colab](https://colab.research.google.com/drive/17KeMdx43KrhLuHV2aTdxRDJE9LbcKr6j?usp=sharing) that builds an example model starting from a pretrained checkpoint ad runs every step I run, including the conversion to FP32 and INT8 at the end. The only difference to my model is the training, the rest is identical.
